utils/build-script

```diff
@@ -740,6 +780,23 @@ the number of parallel build jobs to use""",
                 "--symbols-package.")
             return 1
 
+    if args.android:
+        if args.android_ndk is None or \
+                args.android_ndk_version is None or \
+                args.android_icu_uc is None or \
+                args.android_icu_uc_include is None or \
+                args.android_icu_i18n is None or \
+                args.android_icu_i18n_include is None:
+            print_with_argv0("When building for Android, --android-ndk, "
+                             "--android-ndk-version, --android-icu-uc, "
+                             "--android-icu-uc-include, --android-icu-i18n, "
+                             "and --android-icu-i18n-include must be "
+                             "specified.")
+            return 1
+        # lib/Driver/ToolChains.cpp depends on this environment variable being
```

Comment:
I agree that it shouldn't have to be an environment variable. My comment about being cleaner relates to the fact that all of the above are just subpaths of ANDROID_NDK_HOME, so it seems unnecessary to pass each of them in individually.

I'd support a change to pass NDK_HOME itself into build-script as a variable, and only compile android if it is present. Not sure how this fits in with the logic of deciding which other platforms to build for though